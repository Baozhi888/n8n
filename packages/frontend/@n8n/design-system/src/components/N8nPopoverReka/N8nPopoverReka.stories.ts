import type { StoryFn } from '@storybook/vue3';
import { ref } from 'vue';

import N8nPopoverReka from './N8nPopoverReka.vue';
import N8nButton from '../N8nButton/Button.vue';
import N8nInput from '../N8nInput/Input.vue';

export default {
	title: 'Atoms/PopoverReka',
	component: N8nPopoverReka,
};

const Template: StoryFn = (args) => ({
	setup() {
		const username = ref('');
		const email = ref('');

		return { args, username, email };
	},
	components: {
		N8nPopoverReka,
		N8nButton,
		N8nInput,
	},
	template: `
		<div style="padding: 50px;">
			<N8nPopoverReka v-bind="args">
				<template #trigger>
					<N8nButton type="primary">Open Form</N8nButton>
				</template>
				<template #content>
					<div style="display: flex; flex-direction: column; gap: 12px;">
						<h3 style="margin: 0 0 8px 0; font-size: 14px; font-weight: 600;">User Information</h3>
						<N8nInput
							v-model="username"
							placeholder="Enter username"
							label="Username"
						/>
						<N8nInput
							v-model="email"
							placeholder="Enter email"
							label="Email"
							type="email"
						/>
						<div style="display: flex; gap: 8px; margin-top: 8px;">
							<N8nButton size="small" type="primary">Save</N8nButton>
							<N8nButton size="small" type="secondary">Cancel</N8nButton>
						</div>
					</div>
				</template>
			</N8nPopoverReka>
		</div>
	`,
});

export const SimpleExample = Template.bind({});
SimpleExample.args = {};
SimpleExample.storyName = 'With Form Inputs';
