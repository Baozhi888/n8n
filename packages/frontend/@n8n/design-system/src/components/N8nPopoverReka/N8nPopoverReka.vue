<script setup lang="ts">
import { PopoverContent, PopoverPortal, PopoverRoot, PopoverTrigger } from 'reka-ui';
</script>

<template>
	<PopoverRoot>
		<PopoverTrigger :as-child="true">
			<slot name="trigger"></slot>
		</PopoverTrigger>
		<PopoverPortal>
			<PopoverContent side="bottom" :side-offset="5" :class="$style.popoverContent">
				<slot name="content" />
			</PopoverContent>
		</PopoverPortal>
	</PopoverRoot>
</template>

<style lang="scss" module>
.popoverContent {
	border-radius: var(--border-radius-base);
	padding: 16px;
	width: 260px;
	background-color: var(--color-foreground-xlight);
	border: var(--border-base);
	box-shadow:
		rgba(0, 0, 0, 0.1) 0px 10px 15px -3px,
		rgba(0, 0, 0, 0.05) 0px 4px 6px -2px;
	animation-duration: 400ms;
	animation-timing-function: cubic-bezier(0.16, 1, 0.3, 1);
	will-change: transform, opacity;
}

.popoverContent[data-state='open'][data-side='top'] {
	animation-name: slideDownAndFade;
}

.popoverContent[data-state='open'][data-side='right'] {
	animation-name: slideLeftAndFade;
}

.popoverContent[data-state='open'][data-side='bottom'] {
	animation-name: slideUpAndFade;
}

.popoverContent[data-state='open'][data-side='left'] {
	animation-name: slideRightAndFade;
}

@keyframes slideUpAndFade {
	from {
		opacity: 0;
		transform: translateY(2px);
	}
	to {
		opacity: 1;
		transform: translateY(0);
	}
}

@keyframes slideRightAndFade {
	from {
		opacity: 0;
		transform: translateX(-2px);
	}
	to {
		opacity: 1;
		transform: translateX(0);
	}
}

@keyframes slideDownAndFade {
	from {
		opacity: 0;
		transform: translateY(-2px);
	}
	to {
		opacity: 1;
		transform: translateY(0);
	}
}

@keyframes slideLeftAndFade {
	from {
		opacity: 0;
		transform: translateX(2px);
	}
	to {
		opacity: 1;
		transform: translateX(0);
	}
}
</style>
