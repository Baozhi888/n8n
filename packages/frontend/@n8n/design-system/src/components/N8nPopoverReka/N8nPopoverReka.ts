import type { StoryFn } from '@storybook/vue3';

import N8nPopoverReka from './N8nPopoverReka.vue';

export default {
	title: 'Atoms/N8nPopoverReka',
	component: N8nPopoverReka,
};

const Template: StoryFn = (args) => ({
	setup: () => ({ args }),
	props: args,
	components: {
		N8nPopoverReka,
	},
	template: '<popover-reka v-bind="args" />',
});

export const AllColumnsShown = Template.bind({});
AllColumnsShown.args = {};
