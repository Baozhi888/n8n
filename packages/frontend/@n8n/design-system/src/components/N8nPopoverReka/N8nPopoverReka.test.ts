import { render } from '@testing-library/vue';

import N8nPopoverReka from './N8nPopoverReka.vue';

const defaultStubs = {
	PopoverContent: {
		template: '<mock-popover-content><slot /></mock-popover-content>',
		props: ['open'],
	},
	PopoverPortal: { template: '<mock-popover-portal><slot /></mock-popover-portal>' },
	PopoverRoot: { template: '<mock-popover-root><slot /></mock-popover-root>' },
	PopoverTrigger: { template: '<mock-popover-trigger><slot /></mock-popover-trigger>' },
};

describe('N8nPopoverReka', () => {
	it('should render correctly', () => {
		const wrapper = render(N8nPopoverReka, {
			props: {},
			global: {
				stubs: defaultStubs,
			},
			slots: {
				trigger: '<button />',
				content: '<content />',
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
