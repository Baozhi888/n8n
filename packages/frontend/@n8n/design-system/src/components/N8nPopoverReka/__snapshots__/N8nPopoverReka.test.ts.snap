// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`N8nPopoverReka > should render correctly 1`] = `
"<mock-popover-root>
  <mock-popover-trigger as-child="true"><button></button></mock-popover-trigger>
  <mock-popover-portal>
    <mock-popover-content side="bottom" side-offset="5" class="popoverContent">
      <content></content>
    </mock-popover-content>
  </mock-popover-portal>
</mock-popover-root>"
`;

exports[`N8nPopoverReka > should render correctly with default props 1`] = `
"<mock-popover-root>
  <mock-popover-trigger><button></button></mock-popover-trigger>
  <mock-popover-portal>
    <mock-popover-content>
      <content></content>
    </mock-popover-content>
  </mock-popover-portal>
</mock-popover-root>"
`;

exports[`N8nPopoverReka > should render with open prop set to true 1`] = `
"<mock-popover-root>
  <mock-popover-trigger><button>Open Popover</button></mock-popover-trigger>
  <mock-popover-portal>
    <mock-popover-content>
      <div>Popover Content</div>
    </mock-popover-content>
  </mock-popover-portal>
</mock-popover-root>"
`;
